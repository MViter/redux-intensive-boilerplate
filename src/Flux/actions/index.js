export const changePage = (page) => ({
    type:    'CHANGE_PAGE',
    payload: page
});
