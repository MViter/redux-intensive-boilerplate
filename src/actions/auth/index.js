// Types
import types from './types';

export default {
    login: (user) => ({
        type:    types.LOGIN,
        payload: user
    }),
    loginSuccess: () => ({
        type: types.LOGIN_SUCCESS
    }),
    loginFail: (message) => ({
        type:    types.LOGIN_FAIL,
        payload: message
    }),
    signup: (user) => ({
        type:    types.SIGNUP,
        payload: user
    }),
    signupSuccess: () => ({
        type: types.SIGNUP_SUCCESS
    }),
    signupFail: (message) => ({
        type:    types.SIGNUP_FAIL,
        payload: message
    }),
    logout: () => ({
        type: types.LOGOUT
    }),
    logoutSuccess: () => ({
        type: types.LOGOUT_SUCCESS
    }),
    logoutFail: (message) => ({
        type:    types.LOGOUT_FAIL,
        payload: message
    })
};
