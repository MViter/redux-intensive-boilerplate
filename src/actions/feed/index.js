import types from './types';

export default{
    fetchMovies: (type) => ({
        type:    types.FETCH_MOVIES,
        payload: type
    }),
    fetchMoviesSuccess: (movies) => ({
        type:    types.FETCH_MOVIES_SUCCESS,
        payload: movies
    }),
    fetchMoviesFail: (error) => ({
        type:    types.FETCH_MOVIES_FAIL,
        payload: error
    }),

    fetchGenres: () => ({
        type: types.FETCH_GENRES
    }),
    fetchGenresSuccess: (genres) => ({
        type:    types.FETCH_GENRES_SUCCESS,
        payload: genres
    }),
    fetchGenresFail: (error) => ({
        type:    types.FETCH_GENRES_FAIL,
        payload: error
    })
};
