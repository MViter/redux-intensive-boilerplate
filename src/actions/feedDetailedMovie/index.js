import types from './types';

export default{
    fetchDetailedMovie: (movieId) => ({
        type:    types.FETCH_DETAILED_MOVIE,
        payload: movieId
    }),
    fetchDetailedMovieSuccess: (movie) => ({
        type:    types.FETCH_DETAILED_MOVIE_SUCCESS,
        payload: movie
    }),
    fetchDetailedMovieFail: (error) => ({
        type:    types.FETCH_DETAILED_MOVIE_FAIL,
        payload: error
    }),
    clearDetailedMovie: () => ({
        type: types.CLEAR_DETAILED_MOVIE
    })
};
