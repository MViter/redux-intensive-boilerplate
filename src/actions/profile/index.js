// Types
import types from './types';

export default {
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user
    }),
    updateProfile: (user) => ({
        type:    types.UPDATE_PROFILE,
        payload: user
    }),
    updateProfileSuccess: (user) => ({
        type:    types.UPDATE_PROFILE_SUCCESS,
        payload: user
    }),
    updateProfileFail: (message) => ({
        type:    types.UPDATE_PROFILE_FAIL,
        payload: message
    }),
    updateAvatar: (avatar) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatar
    }),
    updateAvatarSuccess: (url) => ({
        type:    types.UPDATE_AVATAR_SUCCESS,
        payload: url
    }),
    updateAvatarFail: (message) => ({
        type:    types.UPDATE_AVATAR_FAIL,
        payload: message
    })
};
