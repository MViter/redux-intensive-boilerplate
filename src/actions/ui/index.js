// Core
import types from './types';

export default {
    initialize: () => ({
        type: types.INITIALIZE
    }),
    startFetchingAuth: () => ({
        type: types.START_FETCHING_AUTH
    }),
    stopFetchingAuth: () => ({
        type: types.STOP_FETCHING_AUTH
    }),
    startFetchingFeed: () => ({
        type: types.START_FETCHING_FEED
    }),
    stopFetchingFeed: () => ({
        type: types.STOP_FETCHING_FEED
    }),
    startFetchingProfile: () => ({
        type: types.START_FETCHING_PROFILE
    }),
    stopFetchingProfile: () => ({
        type: types.STOP_FETCHING_PROFILE
    }),
    startFetchingAvatar: () => ({
        type: types.START_FETCHING_AVATAR
    }),
    stopFetchingAvatar: () => ({
        type: types.STOP_FETCHING_AVATAR
    }),
    startProfileEditing: () => ({
        type: types.START_PROFILE_EDITING
    }),
    stopProfileEditing: () => ({
        type: types.STOP_PROFILE_EDITING
    }),
    startPasswordEditing: () => ({
        type: types.START_PASSWORD_EDITING
    }),
    stopPasswordEditing: () => ({
        type: types.STOP_PASSWORD_EDITING
    })
};
