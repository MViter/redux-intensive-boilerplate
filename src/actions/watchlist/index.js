import types from './types';

export default{

    fetchWatchlist: () => ({
        type: types.FETCH_WATCHLIST
    }),
    fetchWatchlistSuccess: (movieIDs) => ({
        type:    types.FETCH_WATCHLIST_SUCCESS,
        payload: movieIDs
    }),
    fetchWatchlistFail: (message) => ({
        type:    types.FETCH_WATCHLIST_FAIL,
        payload: message
    }),

    addToWatchlist: (movieToAddId) => ({
        type:    types.ADD_TO_WATCHLIST,
        payload: movieToAddId
    }),
    addToWatchlistSuccess: (movies) => ({
        type:    types.ADD_TO_WATCHLIST_SUCCESS,
        payload: movies
    }),
    addToWatchlistFail: (message) => ({
        type:    types.ADD_TO_WATCHLIST_FAIL,
        payload: message
    }),

    deleteFromWatchlist: (movie) => ({
        type:    types.DELETE_FROM_WATCHLIST,
        payload: movie
    }),
    deleteFromWatchlistSuccess: (movies) => ({
        type:    types.DELETE_FROM_WATCHLIST_SUCCESS,
        payload: movies
    }),
    deleteFromWatchlistFail: (message) => ({
        type:    types.DELETE_FROM_WATCHLIST_FAIL,
        payload: message
    })
};
