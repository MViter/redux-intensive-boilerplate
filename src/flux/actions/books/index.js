export const changePage = (page) => {
    return {
        type:    'CHANGE_PAGE',
        payload: page
    };
};
