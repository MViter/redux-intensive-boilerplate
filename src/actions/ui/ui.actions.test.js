// Core
import actions from './';

describe('ui actions:', () => {
    test(`'initialize' action creator should produce a corresponding action`, () => {
        expect(actions.initialize()).toMatchSnapshot();
    });
    test(`'startFetchingAuth' action creator should produce a corresponding action`, () => {
        expect(actions.startFetchingAuth()).toMatchSnapshot();
    });
    test(`'stopFetchingAuth' action creator should produce a corresponding action`, () => {
        expect(actions.stopFetchingAuth()).toMatchSnapshot();
    });
    test(`'startFetchingFeed' action creator should produce a corresponding action`, () => {
        expect(actions.startFetchingFeed()).toMatchSnapshot();
    });
    test(`'stopFetchingFeed' action creator should produce a corresponding action`, () => {
        expect(actions.stopFetchingFeed()).toMatchSnapshot();
    });
    test(`'startFetchingProfile' action creator should produce a corresponding action`, () => {
        expect(actions.startFetchingProfile()).toMatchSnapshot();
    });
    test(`'stopFetchingProfile' action creator should produce a corresponding action`, () => {
        expect(actions.stopFetchingProfile()).toMatchSnapshot();
    });
    test(`'startFetchingAvatar' action creator should produce a corresponding action`, () => {
        expect(actions.startFetchingAvatar()).toMatchSnapshot();
    });
    test(`'stopFetchingAvatar' action creator should produce a corresponding action`, () => {
        expect(actions.stopFetchingAvatar()).toMatchSnapshot();
    });
    test(`'startProfileEditing' action creator should produce a corresponding action`, () => {
        expect(actions.startProfileEditing()).toMatchSnapshot();
    });
    test(`'stopProfileEditing' action creator should produce a corresponding action`, () => {
        expect(actions.stopProfileEditing()).toMatchSnapshot();
    });
    test(`'startPasswordEditing' action creator should produce a corresponding action`, () => {
        expect(actions.startPasswordEditing()).toMatchSnapshot();
    });
    test(`'stopPasswordEditing' action creator should produce a corresponding action`, () => {
        expect(actions.stopPasswordEditing()).toMatchSnapshot();
    });
});
