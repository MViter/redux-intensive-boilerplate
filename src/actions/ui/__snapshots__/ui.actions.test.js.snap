// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: 'initialize' action creator should produce a corresponding action 1`] = `
Object {
  "type": "INITIALIZE",
}
`;

exports[`ui actions: 'startFetchingAuth' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_FETCHING_AUTH",
}
`;

exports[`ui actions: 'startFetchingAvatar' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_FETCHING_AVATAR",
}
`;

exports[`ui actions: 'startFetchingFeed' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_FETCHING_FEED",
}
`;

exports[`ui actions: 'startFetchingProfile' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_FETCHING_PROFILE",
}
`;

exports[`ui actions: 'startPasswordEditing' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_PASSWORD_EDITING",
}
`;

exports[`ui actions: 'startProfileEditing' action creator should produce a corresponding action 1`] = `
Object {
  "type": "START_PROFILE_EDITING",
}
`;

exports[`ui actions: 'stopFetchingAuth' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_FETCHING_AUTH",
}
`;

exports[`ui actions: 'stopFetchingAvatar' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_FETCHING_AVATAR",
}
`;

exports[`ui actions: 'stopFetchingFeed' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_FETCHING_FEED",
}
`;

exports[`ui actions: 'stopFetchingProfile' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_FETCHING_PROFILE",
}
`;

exports[`ui actions: 'stopPasswordEditing' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_PASSWORD_EDITING",
}
`;

exports[`ui actions: 'stopProfileEditing' action creator should produce a corresponding action 1`] = `
Object {
  "type": "STOP_PROFILE_EDITING",
}
`;
